### PR TITLE
Only generate docs on Pull Requests to Master

### PR DESCRIPTION
**Pull request by sam-heilbron**
# Description

Only pull requests to master (not any other branches) should build our docs to verify that they haven't changed.

# Context

The `master` branch is used to maintain our docs. This means that on pull requests to master we want to check our docs gen. And on pushes to master we want to deploy them.

Previously, we were running build-docs on all PRs, which would fail on backported branches (ie https://github.com/solo-io/gloo/pull/4344/checks?check_run_id=1988513022). 

This change uses the same github workflow to build and publish our docs. This way we have a single place that controls our docs. The rules are: if it is a pull request to master, build the docs, if it is a push to master, build the docs and deploy them!

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

# Additional Testing
- [x] https://github.com/solo-io/gloo/pull/4354 does not run docs gen, since it's not a PR against master

### CODE REVIEW

File: .github/workflows/regression-tests-and-codegen.yaml [hunk 1]
```diff
@@ -31,23 +31,7 @@ jobs:
...
-      shell: bash
-      run: |
-          echo "Pull Request is from a fork. Setting IS_COMMUNITY_PR to true"
-          echo "::set-output name=IS_COMMUNITY_PR::true"
-    - name: Generate versioned docs site
-      run: make -C docs build-site
-      env:
-        GITHUB_TOKEN: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
-        SKIP_CHANGELOG_GENERATION: ${{ steps.community-pr-check.outputs.IS_COMMUNITY_PR }}
+        ./ci/check-generated-code.sh
```
Comment (npolshakova) [inline .github/workflows/regression-tests-and-codegen.yaml:34]: 🎉 nice rename

Comment (EItanya) [reply .github/workflows/regression-tests-and-codegen.yaml:34]: I don't see this file moved, is this a new file?

**Review by EItanya — COMMENTED** [review]
lgtm, the only thing remaining would probably be a workflow to create a PR on master when we do releases to bump `active_versions.json`

File: .github/workflows/regression-tests-and-codegen.yaml (hunk 1 above)
Comment (saiskee) [reply .github/workflows/regression-tests-and-codegen.yaml:34]: image

**Comment by sam-heilbron** [comment]
> lgtm, the only thing remaining would probably be a workflow to create a PR on master when we do releases to bump `active_versions.json`

Agreed, I opened an issue to track that request: https://github.com/solo-io/gloo/issues/4361

**Comment by sam-heilbron** [comment]
/kick https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=8679bb63-cbfc-4d97-8d6f-16ca38f15847

**Review by npolshakova — APPROVED** [review]
Looks good!

**Pull request merged**